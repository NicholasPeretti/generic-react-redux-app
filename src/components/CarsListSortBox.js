import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Style from '../style'
import Dropdown from './Dropdown'
import Text from './Text'

const NONE = 'None'
const MILEAGE_ASC = 'Mileage - Ascending'
const MILEAGE_DESC = 'Mileage - Descending'

const Box = styled.div`
  display: flex;
  flex-direction: column;
`

const StyledText = styled(Text)`
  margin-bottom: ${Style.spacing.padding1};
`

var CarsListSortBox = ({ setSort }) => {
  var listOptions = [
    NONE,
    MILEAGE_ASC,
    MILEAGE_DESC
  ]

  return (
    <Box>
      <StyledText>Sort by</StyledText>
      <Dropdown
        items={listOptions}
        onChange={(val) => {
          let sortValue = (val === NONE) ? null : val
          setSort(sortValue)
        }}
      />
    </Box>
  )
}

CarsListSortBox.propTypes = {
  setSort: PropTypes.func.isRequired
}

export default CarsListSortBox
