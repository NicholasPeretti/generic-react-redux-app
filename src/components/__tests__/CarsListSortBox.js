import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import CarsListSortBox from '../CarsListSortBox'

const props = {
  setSort: jest.fn()
}

describe('CarsListSortBox', () => {
  PresentationalTest(CarsListSortBox, props)

  test('Should match snapshot', () => {
    const wrapper = shallow(<CarsListSortBox {...props} />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
