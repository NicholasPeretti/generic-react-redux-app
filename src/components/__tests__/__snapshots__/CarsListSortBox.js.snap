// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsListSortBox Should match snapshot 1`] = `"<div class=\\"sc-bZQynM AMncx\\"><span class=\\"sc-bdVaJa sc-gzVnrw hlocEI\\">Sort by</span><div class=\\"sc-bwzfXH gYNwqV\\"><button class=\\"sc-htpNat kYidVU\\"><span class=\\"sc-bdVaJa bkmfbd\\">None</span><span class=\\"sc-bxivhb kAOnLj\\"><span>â–¼</span></span></button></div></div>"`;
